eo-maven-plugin/src/main/java/org/eolang/maven/Home.java

```diff
@@ -0,0 +1,189 @@
+     *  we can invent any other solution
+     */
+    Home(final Path path) {
+        this.cwd = path;
+    }
+
+    /**
+     * Saving input.
+     * @param input Input
+     * @param path Path
+     * @return Saved bytes
+     * @throws IOException If fails
+     */
+    public long save(final Input input, final Path path) throws IOException {
+        final File dir = path.toFile().getParentFile();
+        if (dir.mkdirs()) {
+            Logger.debug(
+                this, "Directory created: %s",
+                this.rel(path.getParent())
+            );
+        }
+        try {
+            final long bytes = new IoChecked<>(
+                new LengthOf(
+                    new TeeInput(
+                        input,
+                        new OutputTo(path)
+                    )
+                )
+            ).value();
+            Logger.debug(
+                Home.class, "File %s saved (%.0f bytes)",
+                this.rel(path), bytes
+            );
+            return bytes;
+        } catch (final IOException ex) {
+            throw new IOException(
+                String.format(
+                    "Failed while trying to save to %s",
+                    this.rel(path)
+                ),
+                ex
+            );
+        }
+    }
+
+    /**
+     * Saving string.
+     * @param str String
+     * @param path Path
+     * @return Saved bytes
+     * @throws IOException If fails
+     */
+    public long save(final String str, final Path path) throws IOException {
+        return this.save(new InputOf(str), path);
+    }
+
+    /**
+     * Saving text.
+     * @param txt Text
+     * @param path Path
+     * @return Saved bytes
+     * @throws IOException If fails
+     */
+    public long save(final Text txt, final Path path) throws IOException {
+        return this.save(new InputOf(txt), path);
+    }
+
+    /**
+     * Saving stream.
+     * @param stream Input stream
+     * @param path Path
+     * @return Saved bytes
+     * @throws IOException If fails
+     */
+    public long save(final InputStream stream, final Path path) throws IOException {
+        return this.save(new InputOf(stream), path);
+    }
+
+    /**
+     * Saving bytes.
+     * @param bytes Byte array
+     * @param path Path
+     * @return Saved bytes
+     * @throws IOException If fails
+     */
+    public long save(final byte[] bytes, final Path path) throws IOException {
+        return this.save(new InputOf(bytes), path);
+    }
+
+    /**
+     * Saving string.
+     * @param str String
+     * @param path Path
+     * @return Saved bytes
+     * @todo #1105:30min We should remove this function and get rid of using it
+     *  because it is not really better than simple save
+     */
+    public long saveQuietly(final String str, final Path path) {
+        try {
+            return this.save(new InputOf(str), path);
+        } catch (final IOException ex) {
+            throw new IllegalStateException(ex);
```

Comment:
I don't think it is a good practice to catch one type of exception and throw another one. The stack trace of the original exception is lost.

Reply:
This code is not mine, it was pasted from previous implementation, but I got your point. We want to remove this method at all in future. You may also see `todo`